pkg/crc/preflight/preflight_daemon_task_check_windows.go

```diff
@@ -101,7 +107,7 @@ func removeDaemonTask() error {
 			return err
 		}
 	}
-	if err := checkIfDaemonTaskInstalled(); err == nil {
+	if err := checkIfDaemonTaskInstalled(); err == nil || errors.Is(err, errOlderVersion) {
 		_, stderr, err := powershell.Execute("Unregister-ScheduledTask", "-TaskName", constants.DaemonTaskName, "-Confirm:$false")
 		if err != nil {
 			logging.Debugf("unable to unregister the %s task: %v : %s", constants.DaemonTaskName, err, stderr)
```

Comment:
Fwiw, the full method is
```
if err := checkIfDaemonTaskInstalled(); err == nil || errors.Is(err, errOlderVersion) {
		_, stderr, err := powershell.Execute("Unregister-ScheduledTask", "-TaskName", constants.DaemonTaskName, "-Confirm:$false")
		if err != nil {
			logging.Debugf("unable to unregister the %s task: %v : %s", constants.DaemonTaskName, err, stderr)
			return err
		}
	}
	return nil
}
```
I don't think we should ignore the error from `checkIfDaemonTaskInstalled`, and the return at the end of the method should be `return err`.